quill-engine/src/main/scala/io/getquill/sql/SqlQuery.scala

```diff
@@ -180,24 +186,16 @@ class SqlQueryApply(traceConfig: TraceConfig) {
     nestNextMap: Boolean
   ): FlattenSqlQuery = {
 
-    def select(alias: String, quat: Quat) = SelectValue(Ident(alias, quat), None) :: Nil
+    def select(alias: String, quat: Quat): List[SelectValue] = SelectValue(Ident(alias, quat), None) :: Nil
 
     def base(q: Ast, alias: String, nestNextMap: Boolean): FlattenSqlQuery =
       trace"Computing Base (nestingMaps=${nestNextMap}) for Query: $q" andReturn {
-        def nest(ctx: FromContext) = trace"Computing FlattenSqlQuery for: $ctx" andReturn {
+        def nest(ctx: FromContext): FlattenSqlQuery = trace"Computing FlattenSqlQuery for: $ctx" andReturn {
           FlattenSqlQuery(from = sources :+ ctx, select = select(alias, q.quat))(q.quat)
         }
         q match {
           case _: GroupByMap => trace"base| Nesting GroupByMap $q" andReturn nest(source(q, alias))
           case Map(GroupBy(input, id, _), _, _) =>
-            val innerContext =
-              input match {
-                case _: Map =>
-                  nest(source(input, id.name))
-                case _ =>
-                  base(input, id.name, nestNextMap)
-              }
```

Comment:
@deusaquilus Isn't it weird that this code was here but unused? Could this be a bug: this code should be used in the result? Or is it just dead code we can remove?

Reply:
I think at some point I needed to check whether `input` of a `groupBy.map(...)` is another map i.e. `x.map.groupBy.map` and do a forced nesting but then the nestNextMap took care of that on a higher level.